llvm/test/Verifier/atomics.ll

```diff
@@ -1,14 +1,15 @@
 ; RUN: not opt -passes=verify < %s 2>&1 | FileCheck %s
+; CHECK: atomic store operand must have integer, pointer, floating point, or vector type!
+; CHECK: atomic load operand must have integer, pointer, floating point, or vector type!
 
-; CHECK: atomic store operand must have integer, pointer, or floating point type!
-; CHECK: atomic load operand must have integer, pointer, or floating point type!
+%ty = type { i32 };
 
-define void @foo(ptr %P, <1 x i64> %v) {
-  store atomic <1 x i64> %v, ptr %P unordered, align 8
+define void @foo(ptr %P, %ty %v) {
+  store atomic %ty %v, ptr %P unordered, align 8
   ret void
 }
 
-define <1 x i64> @bar(ptr %P) {
-  %v = load atomic <1 x i64>, ptr %P unordered, align 8
-  ret <1 x i64> %v
+define %ty @bar(ptr %P) {
+  %v = load atomic %ty, ptr %P unordered, align 8
+  ret %ty %v
 }
```

Comment:
please can you add <1 x i64> load/store test coverage some place

Reply:
https://github.com/llvm/llvm-project/blob/fb8e577977fd3c1468d573f8597fb2896e4e7f72/llvm/test/CodeGen/X86/atomic-load-store.ll#L113 Just added it here https://github.com/llvm/llvm-project/pull/120316/commits.

Reply:
Also pointers (including the 32-bit addrspace on x86-64 case which is some high number)